tests-legacy/Unit/Core/Util/YamlParserTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace LegacyTests\Unit\Core\Util\File;
+
+use LegacyTests\TestCase\UnitTestCase;
+use PrestaShop\PrestaShop\Core\Util\File\YamlParser;
+
+class YamlParserTest extends UnitTestCase
+{
+    /**
+     * @param $yamlFiles
+     * @return string
+     */
+    private function clearCacheFile($yamlFiles)
+    {
+        $yamlParser = new YamlParser(false);
+        $cacheFile = $yamlParser->getCacheFile($yamlFiles);
+        @unlink($cacheFile);
+
+        return $cacheFile;
+    }
+
+    public function getConfigDir()
+    {
+        return _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR . 'app' . DIRECTORY_SEPARATOR . 'config';
```

Comment:
Be careful! you must not use the root config dir as your test resources
Especially since you modify them with `touch`, this means running the tests will depreciate the cache for the shop..
Use files located in `tests-legacy/resources` (or `tests/Resources` if this test is moved)

Reply:
Nope, he's just targeting the config directory which is legit here. Otherwise why having a config_test.yml in app/config ? :wink:

Reply:
Yes but in `testParserCache` there is a `touch($yamlFiles, time()+1);` which changes the date of `app/config/config_test.yml`
Ok it's the config test (I admit I first thought it was the config.yml file), but it would be cleaner if these files were test resources

Reply:
Oooohhh, you just changed it! ^^ You're malicious!

Reply:
Yeah, thinking he had forgotten.